src/dinit.cc

```diff
@@ -718,7 +723,19 @@ int dinit_main(int argc, char **argv)
             // Services all stopped but there was no shutdown issued. Inform user, wait for ack, and
             // re-start boot sequence.
             sync(); // Sync to minimise data loss if user elects to power off / hard reset
-            confirm_restart_boot();
+            if (auto_recovery) {
+                try {
+                    log_msg_end(" Will start recovery.");
```

Comment:
`log_msg_end` is supposed to be paired with `log_msg_begin` (and zero or more `log_msg_part`s). To issue a message that doesn't consist of multiple parts you can use a single call to `log` instead.

The message `" Will start recovery."` is not self-contained (and starts with a space). I think maybe you copied and modified from one of the lines above, which are all part of a two-part message. If you log a message, it should be self-contained and should not start with white space. I would suggest just `"Starting recovery."`.

Reply:
The space was to account for https://github.com/davmac314/dinit/blob/927bec0/src/dinit.cc#L686-L697 not handling `shutdown_type_t::NONE` as I saw the log contained the next thing on the same line after `No more active services.` without new line, should that be dealt with or left as-is?

Reply:
(Ok, this is fine, see my other comment).